src/docs/sphinx/dev_guide/release.rst

```diff
@@ -15,66 +15,65 @@ e.g., when some new functionality is sufficiently complete or we want users to
 try something out and give us feedback early in the development process. A
 release may also be done when some other development goal is reached. This
 section describes how an Axom releases is done. The process is fairly
-informal. However, we want to ensure that the software is in a reasonably
-robust and stable state when a release is done. We follow this process to
-avoid simple oversights and issues that we do not want to pass on to users.
+informal. However, we want to ensure that the software is in a robust and
+stable state when a release is done. We follow the process described in this
+section to avoid oversights and issues that we do not want to pass on to users.
 
 In the :ref:`gitflow-label` section, we noted that the **main branch
 records the official release history of the project**. Specifically,
 whenever, the main branch is changed, it is tagged with a new
 version number. We use a git 'lightweight tag' for this purpose. Such
 a tag is essentially a pointer to a specific commit on the main branch.
 
-We finalize preparations for a release on a release branch so that other
-work may continue on the develop branch without interruption.
+When all pull requests intended to be included in a release have been merged
+into the develop branch, we create a release candidate branch 
+**from the develop branch**. The release candidate branch is used to finalize 
+preparations for the release. At this point, the next release cycle begins and
+work may continue on the develop branch.
 
-.. note:: No significant code development is performed on a release branch.
-          In addition to preparing release notes and other documentation, the
-          only code changes that should be done are bug fixes identified
-          during release preparations
+.. important:: No significant code development should performed on a release
+               candidate branch. Apart from finalizing release notes and other
+               documentation, **the only code changes** that should be merged
+               into a release candidate branch are bug fixes identified during
+               release preparations. Changing code in a release candidate is
```

Comment:
Don't we change the version number in the release candidate branch?

Reply:
Yes we do. That's notes in the steps for the RC branch. I was thinking that was a documentation change, not a code change. If it's not clear, please suggest a change. The point of this PR is to clarify the process.